src/zimbraweb/emlparsing.py

```diff
@@ -0,0 +1,91 @@
+from typing import Tuple
+import re
+import base64
+
+from email.parser import Parser
+from zimbraweb import ZimbraUser, WebkitAttachment  # for parsing eml
+
+
+def eml_parsing(user: ZimbraUser, eml: str) -> Tuple[bytes, str]:
+    """Generate a payload from any eml
+
+        Parameters:
+            user: ZimbraUser Class
+            eml: str
+
+        Returns:
+            bytes: The WebkitFormBoundary Payload
+            str: The boundary used in the payload
+    """
+
+    parser = Parser()
+    parsed = parser.parsestr(eml)
+
+    if type(parsed.get_payload()) == str:
+        return user.generate_webkit_payload(parsed['To'], parsed['Subject'], parsed.get_payload())
+
+    elif type(parsed.get_payload()) == list:
+        dict_mail = {}
+        dict_mail['to'] = parsed['To']
+        dict_mail['subject'] = parsed['Subject']
+        dict_mail['attachments'] = []
+        return user.generate_webkit_payload(**multipart_eml_parsing(user, parsed.get_payload(), dict_mail))
+
+    else:
+        raise TypeError("Multipart Payload in Plain Parser. Please use multipart_eml_parsing")
+
+
+def plain_eml_parsing(user: ZimbraUser, eml: str) -> Tuple[bytes, str]:
+    # right now this function is useless, since its so short and basically needs to be done for eml_parsing function anyway
+    """Generate a payload from plaintext eml
+
+        Parameters:
+            user: ZimbraUser Class
+            eml: str
+
+        Returns:
+            bytes: The WebkitFormBoundary Payload
+            str: The boundary used in the payload
+    """
+
+    parser = Parser()
+    parsed = parser.parsestr(eml)
+
+    # this if is not strictly necessary, but prevents from calling the plain function for multipart messages
+    if type(parsed.get_payload()) == str:
+        return user.generate_webkit_payload(parsed['To'], parsed['Subject'], parsed.get_payload())
+    else:
+        raise TypeError("Multipart Payload in Plain Parser. Please use multipart_eml_parsing")
+
+
+def multipart_eml_parsing(user: ZimbraUser, payload: list, dict_mail: dict) -> dict:
+    # TODO correct the types, its not a list but a list of email message objects
+    """Generate a dictionary of multipart-parts form a multipart email payload
+
+        Parameters:
+            parsedcontent (dict): an existing dictionary, key 'body' contains the body, key 'attachments' contains a [WebkitAttachments] list of attachments
+            payload (list): list of email message objects
+
```

Comment:
These parameters don't match up with the function signature. Maybe you also need 

user (ZimbraUser): The Zimbra user which sent the eml file.